ui-tests/src/test/java/org/openmrs/reference/DuplicatePatientRegisterTest.java

```diff
@@ -1,74 +1,71 @@
 
+import org.hamcrest.Matchers;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.openmrs.reference.groups.BuildTests;
 import org.openmrs.reference.helper.PatientGenerator;
 import org.openmrs.reference.helper.TestPatient;
+import org.openmrs.reference.page.ClinicianFacingPatientDashboardPage;
 import org.openmrs.reference.page.HeaderPage;
 import org.openmrs.reference.page.HomePage;
 import org.openmrs.reference.page.RegistrationPage;
 import org.openmrs.uitestframework.test.TestBase;
 
 import java.awt.*;
 
+import static org.junit.Assert.assertThat;
 import static org.junit.Assert.assertTrue;
 
 
-public class DuplicatePatientRegisterTest  extends TestBase {
-    private HeaderPage headerPage;
-    private RegistrationPage registrationPage;
-    private HomePage homePage;
+public class DuplicatePatientRegisterTest  extends ReferenceApplicationTestBase {
     private TestPatient patient1;
 
     @Before
     public void setUp() throws AWTException {
-        headerPage = new HeaderPage(driver);
         patient1 = PatientGenerator.generateTestPatient();
-
-        assertPage(page);
-        
-        homePage = new HomePage(page);
-        registrationPage = new RegistrationPage(page);
-        assertPage(homePage);
     }
 
-    private void registerAPatient(TestPatient patient) throws InterruptedException {
-
-        homePage.goToRegisterPatientApp();
-        patient.familyName = "Smith";
-        patient.middleName = "";
-        patient.givenName = "Bob";
-        patient.gender = "Male";
-        patient.birthYear = "1999";
-        patient.birthMonth = "January";
-        patient.birthDay = "1";
-        patient.address1 = "address";
-        patient.address2 = "address";
-        registrationPage.enterPatient(patient);
+    // Test for RA-714
+    @Test
+    @Category(BuildTests.class)
+    public void duplicateRegisterTest() throws InterruptedException {
+        RegistrationPage registrationPage = homePage.goToRegisterPatientApp();
+        registrationPage.enterPatient(patient1);
+        ClinicianFacingPatientDashboardPage dashboardPage = registrationPage.confirmPatient();
+
+        patient1.uuid = dashboardPage.getPatientUuidFromUrl();
+
+        dashboardPage.goToPage("/referenceapplication/home.page");
+
+        registrationPage = homePage.goToRegisterPatientApp();
+        registrationPage.enterPatient(patient1);
+
+        String name = registrationPage.getSimilarPatientName();
+        assertThat(name, Matchers.is(patient1.givenName + " " + patient1.familyName));
+
+        String info = registrationPage.getSimilarPatientInfo();
+        assertThat(info, Matchers.is(patient1.gender + ", " + "0" + patient1.birthDay + "." + patient1.birthMonth.substring(0,3) + "." + patient1.birthYear + ", " + patient1.address1 + " " + patient1.address2 + " " + patient1.city + patient1.state + patient1.country + patient1.postalCode));
```

Comment:
Appending 0 to birthDay doesn't seem bulletproof. How about you use SimpleDateFormat instead to properly format date?